tensorflow_probability/python/mcmc/transformed_kernel.py

```diff
@@ -132,6 +155,26 @@ class TransformedTransitionKernel(kernel_base.TransitionKernel):
   `tfp.mcmc.HamiltonianMonteCarlo`, `tfp.mcmc.RandomWalkMetropolis`,
   etc.
 
+  If the provided `inner_kernel` doesn't have a `target_log_prob_fn`,
```

Comment:
I think this information should be less prominent in the docstring.  It would be better as its own section called something like "Detailed API Contract" or something, probably near the end, between the examples and the references.  It should also be more complete: say at the beginning that `TransformedTransitionKernel` operates on a `target_log_prob_fn` field, say explicitly that it looks for `inner_kernel` fields, etc. The goal for technical documentation like this is to make it so that someone who carefully reads the docstring but doesn't read the code can implement an end kernel or a wrapper kernel that will be transformed correctly.

Reply:
Moved and added more relevant information.